### ISSUE TITLE
could not find function ".rs.checkBuildTools"

### ISSUE THREAD

**Issue by littlehifive** [lindeloev/job#7]
Hi thanks for developing the package! 

The example brm model runs well in console. However, when I ran the example codes to get it in jobs: 
```
# Send long-running code to a job
job::job(brm_result = {
  fit = brm(model, data)
  print(summary(fit))  # Show a summary in the job
})
```
I encountered this error:
```
Compiling Stan program...
Error in .rs.checkBuildTools(action) : 
  could not find function ".rs.checkBuildTools"
Calls: sourceWithProgress ... eval -> .fun -> cxxfunctionplus -> <Anonymous> -> check
Execution halted
```
Let me know how I should deal with this error. Thank you so much!

**Comment by lindeloev** [comment]
This seems related to #4. Unfortunately, I cannot replicate your error. I just pushed a new update which may fix this. Try installing from github again and let me know if it works?

If not, could you reply with `sessionInfo()` and the result of `getOption("buildtools.check")` both in your base console and from within a job with/without `opts = NULL`?

**Comment by littlehifive** [comment]
Sorry I still got the same error after reinstalling the package. Here is my session info:

```
> sessionInfo()
R version 4.0.2 (2020-06-22)
Platform: x86_64-apple-darwin17.0 (64-bit)
Running under: macOS  10.16

Matrix products: default
LAPACK: /Library/Frameworks/R.framework/Versions/4.0/Resources/lib/libRlapack.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] brms_2.15.0 Rcpp_1.0.6 

loaded via a namespace (and not attached):
  [1] readxl_1.3.1         backports_1.2.1      Hmisc_4.5-0         
  [4] igraph_1.2.6         plyr_1.8.6           repr_1.1.0          
  [7] sp_1.4-2             splines_4.0.2        crosstalk_1.1.0.1   
 [10] TH.data_1.0-10       ggplot2_3.3.3        inline_0.3.17       
 [13] rstantools_2.1.1     digest_0.6.27        htmltools_0.5.1.1   
 [16] rsconnect_0.8.16     fansi_0.4.2          magrittr_2.0.1      
 [19] checkmate_2.0.0      cluster_2.1.0        openxlsx_4.2.3      
 [22] remotes_2.2.0        RcppParallel_5.1.2   matrixStats_0.56.0  
 [25] sandwich_2.5-1       xts_0.12-0           prettyunits_1.1.1   
 [28] jpeg_0.1-8.1         colorspace_2.0-0     skimr_2.1.2         
 [31] blob_1.2.1           haven_2.4.0          xfun_0.22           
 [34] dplyr_1.0.5          callr_3.4.4          crayon_1.4.1        
 [37] jsonlite_1.7.2       lme4_1.1-26          survival_3.1-12     
 [40] zoo_1.8-8            glue_1.4.2           gtable_0.3.0        
 [43] emmeans_1.4.8        V8_3.4.0             mi_1.0              
 [46] car_3.0-8            pkgbuild_1.1.0       rstan_2.21.2        
 [49] DEoptimR_1.0-8       abind_1.4-5          VIM_6.1.0           
 [52] scales_1.1.1         mvtnorm_1.1-1        DBI_1.1.0           
 [55] miniUI_0.1.1.1       xtable_1.8-4         laeken_0.5.1        
 [58] htmlTable_2.1.0      foreign_0.8-80       Formula_1.2-4       
 [61] StanHeaders_2.21.0-7 DT_0.14              stats4_4.0.2        
 [64] vcd_1.4-8            htmlwidgets_1.5.3    httr_1.4.2          
 [67] threejs_0.3.3        RColorBrewer_1.1-2   ellipsis_0.3.1      
 [70] mice_3.13.0          pkgconfig_2.0.3      loo_2.4.1           
 [73] nnet_7.3-14          utf8_1.2.1           reshape2_1.4.4      
 [76] tidyselect_1.1.0     rlang_0.4.10         later_1.1.0.1       
 [79] munsell_0.5.0        cellranger_1.1.0     tools_4.0.2         
 [82] cli_2.4.0            generics_0.1.0       ranger_0.12.1       
 [85] broom_0.7.6          ggridges_0.5.2       evaluate_0.14       
 [88] stringr_1.4.0        fastmap_1.1.0        arm_1.11-2          
 [91] yaml_2.2.1           processx_3.4.4       knitr_1.33          
 [94] zip_2.1.1            robustbase_0.93-6    purrr_0.3.4         
 [97] nlme_3.1-148         mime_0.10            projpred_2.0.2      
[100] shinythemes_1.2.0    compiler_4.0.2       bayesplot_1.8.0     
[103] rstudioapi_0.13      curl_4.3             gamm4_0.2-6         
[106] png_0.1-7            e1071_1.7-4          job_0.1             
[109] tibble_3.1.1         statmod_1.4.35       stringi_1.5.3       
[112] ps_1.3.4             Brobdingnag_1.2-6    forcats_0.5.1       
[115] lattice_0.20-41      texreg_1.37.5        Matrix_1.2-18       
[118] markdown_1.1         nloptr_1.2.2.2       shinyjs_2.0.0       
[121] vctrs_0.3.7          pillar_1.6.0         lifecycle_1.0.0     
[124] lmtest_0.9-37        bridgesampling_1.0-0 estimability_1.3    
[127] data.table_1.14.0    httpuv_1.5.4         R6_2.5.0            
[130] latticeExtra_0.6-29  promises_1.1.1       gridExtra_2.3       
[133] rio_0.5.16           codetools_0.2-16     gtools_3.8.2        
[136] colourpicker_1.1.0   boot_1.3-25          MASS_7.3-51.6       
[139] assertthat_0.2.1     rprojroot_2.0.2      withr_2.4.2         
[142] shinystan_2.5.0      multcomp_1.4-16      mgcv_1.8-31         
[145] parallel_4.0.2       hms_1.0.0            grid_4.0.2          
[148] rpart_4.1-15         tidyr_1.1.3          coda_0.19-4         
[151] class_7.3-17         minqa_1.2.4          rmarkdown_2.7       
[154] carData_3.0-4        shiny_1.5.0          base64enc_0.1-3     
[157] dygraphs_1.1.1.6    
```

Running from base console:
```
getOption("buildtools.check")
function (action) 
{
    .rs.checkBuildTools(action)
}
<environment: 0x7f9097e1d880>
```

Do you mean the following? sorry I am not sure if these functions should be used this way.
```
job::job({
  getOption("buildtools.check")
  opts = NULL
})

job::job({
  getOption("buildtools.check")
})
```

If so, both give the following message:
```
2021-04-26 22:07:17: Job started.
> 
> 
2021-04-26 22:07:17: Job finished.
```

**Comment by lindeloev** [comment]
Thanks for the help tracking this down. Sorry I wasn't clear. Try running

    library("brms")
    job::job(brm_result = {
      fit = brm(mpg ~ cyl, mtcars)
      print(summary(fit))  # Show a summary in the job
    }, opts = NULL)

This runs the job session without importing anything from the RStudio session. If this works, then try running the same but with `opts = list(buildtools.check = NULL, buildtools.with = NULL)`. Does this work or at least result in a different error?

**Comment by lindeloev** [comment]
The recent commits addressed this and closed #10. Could you install again and see if it runs now, @littlehifive ?

**Comment by littlehifive** [comment]
Yes it ran without the error. Thank you!
